src/views/PianoRoll/Main.lua

```diff
@@ -0,0 +1,118 @@
+---utility functions---
+
+function GetGlobalTimer()
```

Comment:
The issue with that is that `Memory.update()` is called only on `at_vi()` and some other arbitrary points, which leads to outdated values in some of my code paths.  
I think this could either be resolved by providing some form of a generalized `Memory.get_immediate(...)`  function, or by specifying more clearly _when_ memory is updated, and making it such that it is sufficient for the PianoRoll use cases.

Reply:
I can't debug this right now, but that sounds quite inconsistent with the rest of the script.

Your code which calls `GetGlobalTimer` is reached exclusively during `atdrawd2d`, and thus has no synchronization with any event in the emulator's core. 
That means reading data during the drawing section can fetch unexpected values, for example, very rarely, in the middle of frames and is therefore a bad idea.

The rest of SM64Lua is not sensitive to this as it doesn't read memory while drawing, but it also assumes that the current `Memory` data might not be the latest (as that also can't be guaranteed due to the call order of `atdrawd2d` and `atvi`).

As such, memory-related tasks should be done and saved in core callbacks like `atvi`/`atinput` and not during drawing.

Reply:
> Your code which calls `GetGlobalTimer` is reached exclusively during `atdrawd2d`

This is not true.  
`__clsPianoRoll:jumpTo` registers an `emu.atinput` callback that needs the global timer value, as does `CurrentPianoRollOverride` (which is called from the `PianoRoll.lua` _processor_ file), so both of those should actually read the memory directly, since the main `atinput` callback doesn't update memory, as far as I can see anyways. (And `atvi` isn't necessarily called because of fast forward, I think?)

I understand how the disambiguation is necessary though. I will try modifying the GUI code to refer to `Memory.current.mario_global_timer` instead. (Why is it called `mario_global_timer` anyways? It's not like the Mario object or Mario state are at all involved with this value lol.)

Reply:
Done as appropriate in fe88956683acc0fa48e203f0e84ff553005a67cb.